### PR TITLE
Config with pure python

### PR DESCRIPTION
**Pull request by icemac**
* Add support for Python 3.7, 3.8 and 3.9.
* Drop support for Python 3.4.
* Lint the code.
* Create an own `demo` package as the one in `zope.generations` is no longer compatible with our expectations.

### CODE REVIEW

File: src/zope/app/generations/demo/__init__.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# FOR A PARTICULAR PURPOSE.
+#
+##############################################################################
+"""Demo package for evolution scripts
+
+The evolution scripts in this package are pretty dumb. The just call
+the evolve function defined here with a generation number.
+
+"""
+key = 'zope.generations.demo-generation'
```
Comment (mgedmin) [inline src/zope/app/generations/demo/__init__.py]: I wonder if this should also be `zope.app.generations.demo-generation`?

**Review by mgedmin — APPROVED** [review]
LGTM

File: src/zope/app/generations/demo/__init__.py (hunk 1 above)
Comment (icemac) [reply src/zope/app/generations/demo/__init__.py]: Of course. Thank you for catching it.

**Comment by icemac** [comment]
Thank you for reviewing this PR. 😃

**Pull request merged**